app/src/main/kotlin/org/fossify/messages/activities/ThreadActivity.kt

```diff
@@ -229,7 +115,19 @@ class ThreadActivity : SimpleActivity() {
 
     private val binding by viewBinding(ActivityThreadBinding::inflate)
 
-    override fun onNewIntent(intent: Intent) {
+    override fun onNewIntent(@SuppressLint("UnsafeIntentLaunch") intent: Intent) {
+        // avoid unsafe launch
+        val sanitizer = IntentSanitizer.Builder()
+            .allowPackage(BuildConfig.APPLICATION_ID)
+            .build()
+        try {
+            sanitizer.sanitizeByThrowing(intent)
+        } catch (e: Exception) {
+            toast(e.message ?: getString(org.fossify.commons.R.string.unknown_error_occurred))
```

Comment:
Just use the `showErrorToast(e)` extension like everywhere else. Better exception handling is planned but this is not the way.